addon/-legacy-private/system/relationships/state/has-many.js

```diff
@@ -16,25 +17,36 @@ export default class ManyRelationship extends Relationship {
     // we create a new many array, but in the interim it will be updated if
     // inverse internal models are unloaded.
     this._retainedManyArray = null;
-    this._loadingPromise = null;
+    this._promiseProxy = null;
     this._willUpdateManyArray = false;
     this._pendingManyArrayUpdates = null;
   }
 
-  _updateLoadingPromise(promise, content) {
-    if (this._loadingPromise) {
-      if (content) {
-        this._loadingPromise.set('content', content);
-      }
-      this._loadingPromise.set('promise', promise);
-    } else {
-      this._loadingPromise = PromiseManyArray.create({
-        promise,
-        content,
-      });
+  get currentState() {
+    return this.members.list;
+  }
+
+  get hasRelatedResources() {
+    // check currentState for unloaded records
+    let hasEmptyRecords = this.currentState.reduce((hasEmptyModel, i) => {
```

Comment:
Would this be cleaner to write as a find/filter?

Reply:
unsure, I find I have to similarly pause and reason for any of the three approaches. Happy to embrace another pattern if there's one you prefer,